mountpoint-s3-fs/examples/prefetch_benchmark.rs

```diff
@@ -176,10 +163,31 @@ fn main() {
         let received_size = received_bytes.load(Ordering::SeqCst);
         println!(
             "{}: received {} bytes in {:.2}s: {:.2} Gib/s",
-            i,
+            iteration,
             received_size,
             elapsed.as_secs_f64(),
             (received_size as f64) / elapsed.as_secs_f64() / (1024 * 1024 * 1024 / 8) as f64
         );
     }
 }
+
+fn make_s3_client_from_args(args: &CliArgs) -> Arc<S3CrtClient> {
+    let initial_read_window_size = 1024 * 1024 + 128 * 1024;
+    let mut client_config = S3ClientConfig::new()
+        .read_backpressure(true)
+        .initial_read_window(initial_read_window_size)
+        .endpoint_config(EndpointConfig::new(args.region.as_str()));
+    if let Some(throughput_target_gbps) = args.maximum_throughput_gbps {
+        client_config = client_config.throughput_target_gbps(throughput_target_gbps as f64);
+    }
+    if let Some(limit_gib) = args.crt_memory_limit_gib {
+        client_config = client_config.memory_limit_in_bytes(limit_gib * 1024 * 1024 * 1024);
+    }
+    if let Some(part_size) = args.part_size {
+        client_config = client_config.part_size(part_size as usize);
+    }
+    if let Some(interfaces) = &args.bind {
+        client_config = client_config.network_interface_names(interfaces.clone());
+    }
+    Arc::new(S3CrtClient::new(client_config).unwrap())
```

Comment:
I'd rather propagate the error here and restore the `expect` at the call site.

Reply:
Could also have returned anyhow here. We should make that error public, in any case!